### DISCUSSION TITLE
Dietpi-Software | Readarr

### DISCUSSION THREAD

**Discussion by Maggus101** [MichaIng/DietPi#5004] [Software Request] [answered]
#### Formal software information
- Software title | Readarr
- Short description | Readarr is a ebook collection manager for Usenet and BitTorrent users. It can monitor multiple RSS feeds for new books from your favorite authors and will interface with clients and indexers to grab, sort, and rename them. 
- Official URL (if available) | https://readarr.com/
- GitHub URL (if available) | https://github.com/Readarr/Readarr
- (Official) install documentation | https://github.com/Readarr/Readarr#wiki-linux

#### Are there similar/alternative software titles available with DietPi-Software?
Similar to Radarr V3 in design / implementation. Different purpose

#### What makes your requested software better than the above solutions, if available?
N/A

#### How can DietPi make the installation easier or compatible, than following the install instructions or do APT installation, if available?
Same as Radarr

#### Can you provide the installation steps that you would suggest DietPi-Software to do?
Same as Radarr

#### Are you willing to help maintaining the software installation, e.g. in case of needed setup changes due to updates etc.? This is not needed, but could speed up our decision to implement it, as man power is always a topic :wink:.
Totally inexperienced sorry!

**Comment by Joulinar** [comment]
Many thanks for your request. I moved it to discussions where we start to collect software requests now. Users can vote here, which is one benefit, since FeatHub is sadly down, respectively unmaintained and cannot be used anymore.

**Comment by TommySharpNZ** [comment]
Support for this would be great!

**Comment by MichaIng** [comment]
Btw, for voting use the arrow up button on the first post instead of thumbs reaction.

To keep record: This software title got 13 votes in ~8 months on FeatHub: https://feathub.com/MichaIng/DietPi/+221

**Comment by Joulinar** [comment, marked as answer]
will be part of DietPi v8.7 https://github.com/MichaIng/DietPi/pull/5616

**Comment by roni-estein** [comment]
Hey, this might be something to consider for the installer or perhaps something that falls under documentation only, but readarr has a difference in kind from radarr and sonarr where if you collect both audio and written books you need two separate instances of readarr running. There are further optimizations I've just found where people have the libraries sync'd where you would add it to one readarr instance and the other would add it as well, however I've never heard of that until this moment looking for the documentation on how to install a second instance on the dietpi without docker.

**Reply by MichaIng** [reply]
> if you collect both audio and written books you need two separate instances of readarr running

Do you have a reference for this? Sounds wrong or something worth addressing on Readarr side. ... see it now in readme:
> If you want both an audiobook and ebook of a given book you will need multiple instances.

Sounds like there is room for improvement then. However, this should do it:
```sh
# Create new data directory
mkdir /mnt/dietpi_userdata/readarr2
# Setup logging to /var/log (optional)
rm -Rf /mnt/dietpi_userdata/readarr2/logs*
ln -s /var/log/readarr2 /mnt/dietpi_userdata/readarr2/logs
ln -s /var/log/readarr2/logs.db /mnt/dietpi_userdata/readarr2/logs.db
ln -s /var/log/readarr2/logs.db-shm /mnt/dietpi_userdata/readarr2/logs.db-shm
ln -s /var/log/readarr2/logs.db-wal /mnt/dietpi_userdata/readarr2/logs.db-wal
# Copy service
cp /etc/systemd/system/readarr.service /etc/systemd/system/readarr2.service
# Adjust new service to use new data directory
G_CONFIG_INJECT 'Description=' 'Description=Readarr2' /etc/systemd/system/readarr2.service
G_CONFIG_INJECT 'SyslogIdentifier=' 'SyslogIdentifier=Readarr2' /etc/systemd/system/readarr2.service
G_CONFIG_INJECT 'LogsDirectory=' 'LogsDirectory=readarr2' /etc/systemd/system/readarr2.service
sed -i 's|/mnt/dietpi_userdata/readarr$|/mnt/dietpi_userdata/readarr2|' /etc/systemd/system/readarr2.service
# Enable and start new service
systemctl daemon-reload
systemctl enable --now readarr2
```
Of course you can give it a more meaningful name.

EDIT: Ah, you need to run it on a different TCP port, of course. Not sure whether Readarr has a command-line option for this? Otherwise you need to do it via web interface, but only one of both can run at a time until ports are different.

**Reply by roni-estein** [reply]
Hi thanks for the quick response but yes, though it does look like you have found it here is the reference just in case. I should have added it in the original post. And yes it would be great if it wasn't needed in this way.

[reasoning on multiple instances](https://wiki.servarr.com/readarr/installation#multiple-instances)

I've mostly seen it as readarr-audio but someone suggested speakarr which sounds much more elegant. As a container suggestion.

Where would you configure the new IP or would you need to do that directly from the web interface? In that case do you first need to stop the original readarr service?

Thanks again.

**Reply by MichaIng** [reply]
A container is definitely not needed, the same Readarr installation/binaries can be used unlimited times, but you just need to make each instance/call use a dedicated data directory and different network port. The positive side effect is that if you update Readarr through any interface, it is updated for all interfaces. The only thing that should be done is hook in a post upgrade script to restart all other instances afterwards. I think Servarr applications have such possibility, isn't it?

Ah good, so official docs do it the same way. The steps above preserve the difference install dir (`/opt/readarr` vs `/opt/Readarr`) and data directory, some additional hardenings to the service, and imply the optional changed logs location. But its the same basic approach.

**Reply by roni-estein** [reply]
Thanks I'll give it a try. It might be worth adding it as an option on the installer, or as a section on the docs area. Though I feel like the first person bringing it up, any person I actually know who has readarr installed has the dual instance issue and curses the outset. Most people revert to docker for ease. I just stumbled onto dietpi, and it seems so much lighter and smoother running.

Thanks for all your effort.

**Reply by MichaIng** [reply]
Thanks for the kind feedback. A documentation section is then definitely worth it. Before implementing it as install option, I'd first check whether there are any aims to allow downloading both with a single instance. I cannot imagine that this is so hard to achieve 😄: https://github.com/MichaIng/DietPi-Docs/issues/845
